### PR TITLE
Add migration page to manual

### PR DESCRIPTION
**Pull request by rgrinberg**
This includes the old migration page + a simple migration check list. @diml I've included your sections to the check list

### CODE REVIEW

File: doc/migration.rst [hunk 1]
```diff
@@ -114,3 +70,88 @@ it is not preserved for `dune-workspace` files. I.e. a
...
+---------------
+
+``${foo} and $(foo)`` are no longer valid variable syntax in dune files.
+Variables are defined as ``%{foo}``. This change is done to simplify
+interoperability with bash commands which also use teh ``${foo}`` syntax.
+
+``(files_recursively_in ..)`` is removed
+----------------------------------------
+
+The ``files_recursively_in`` stanza is invalid in dune files. A
```
Comment (diml) [inline doc/migration.rst]: `stanza` is for toplevel things. This should be called "dependency specification".

File: doc/migration.rst [hunk 2]
```diff
@@ -114,3 +70,88 @@ it is not preserved for `dune-workspace` files. I.e. a
...
+:ref:`source_tree <source_tree>` stanza has been introduced to reflect the
+actual function of this stanza.
+
+Escape Sequences
+----------------
+
+Invalid escape sequences such as ``\999`` are not tolerated in quoted strings in
```
Comment (diml) [inline doc/migration.rst]: There were not accepted in jbuild files either, only in block comments. One thing that we reject now and didn't before is `\ `, i.e. `\` followed by a character other that 0-9, x, n, r, t, b.

**Comment by diml** [comment]
Ok, looks good. I prefer to have the timeline first though, to give an overview of what will happen, followed by the concrete details.

**Comment by rgrinberg** [comment]
@diml okay, i addressed your point and added a table for renamed variables as well.

File: doc/migration.rst [hunk 3]
```diff
@@ -114,3 +131,42 @@ it is not preserved for `dune-workspace` files. I.e. a
...
+
+The ``files_recursively_in`` dependency specification is invalid in dune files.
+A :ref:`source_tree <source_tree>` stanza has been introduced to reflect the
+actual function of this stanza.
+
+Escape Sequences
+----------------
+
+Invalid escape sequences of the form ``\x`` where ``x`` is a character other
+than ``[0-9]``, ``x``, ``n``, ``r``, ``t``, ``b``.
```
Comment (diml) [inline doc/migration.rst]: it looks like this sentence is not finished

Comment (rgrinberg) [reply doc/migration.rst]: Oops, finished that.

**Comment by diml** [comment]
BTW, we should keep a migration file or a note in the readme pointing to the migration section of the manual.

**Pull request merged**